File: frontend/src/features/customer/models/CustomerResponseModel.ts
import type { CustomerPhoneNumber } from "./CustomerPhoneNumber";

export interface CustomerResponseModel {
  customerId: string;
  firstName: string;
  lastName: string;

  phoneNumbers: CustomerPhoneNumber[];

  streetAddress: string;
  city: string;
  province: string;
  country: string;
  postalCode: string;
  userId: string;
  isActive?: boolean;
}

File: backend/src/main/java/com/profroid/profroidapp/customersubdomain/presentationLayer/CustomerResponseModel.java
package com.profroid.profroidapp.customersubdomain.presentationLayer;

import com.profroid.profroidapp.customersubdomain.dataAccessLayer.CustomerPhoneNumber;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CustomerResponseModel {
    String customerId;
    String firstName;
    String lastName;
    String email;
    String password;
    List<CustomerPhoneNumber> phoneNumbers;

    String streetAddress;
    String city;
    String province;
    String country;
    String postalCode;
}
